### PR TITLE
feat: use kustomize to get labels back on generated resources

### PR DESCRIPTION
**Pull request by erikgb**
## Description

This PR suggests a light introduction of kustomize, without breaking changes, to reintroduce labels (lost by previous PRs) on generated resources.

## Related issues
- Close https://github.com/aquasecurity/trivy-operator/issues/318
- Close https://github.com/aquasecurity/trivy-operator/pull/325

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by chen-keinan** [comment]
@erikgb open an issue for this PR.

File: Makefile [hunk 1]
```diff
@@ -120,29 +120,37 @@ LOCALBIN ?= $(shell pwd)/bin
...
+$(KUSTOMIZE): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) go install sigs.k8s.io/kustomize/kustomize/v4@$(KUSTOMIZE_VERSION)
+
 .PHONY: verify-generated
 verify-generated: generate-all
 	./hack/verify-generated.sh
 
 .PHONY: generate
 generate: controller-gen
-	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./pkg/..." +rbac:roleName=trivy-operator output:rbac:artifacts:config=deploy/helm/generated
```
Comment (erikgb) [inline Makefile:138]: This is correcting a minor bug from earlier; the rbac-flag was added on the wrong invocation of controller-gen. The clusterrole is a manifest, so should generate the clusterrole from make target manifests.

**Comment by chen-keinan** [comment]
@erikgb can we close this PR ?

**Comment by erikgb** [comment]
> @erikgb can we close this PR ? 

@chen-keinan Why? You could at least give me an explanation, I think.

**Comment by chen-keinan** [comment]
@erikgb not sure what is the status of it , can you elaborate, I see its open for long time , does it depend on something specific and do we need it ?

**Comment by erikgb** [comment]
The motivation is described in the issue. It has been ready for review for a long time.

**Comment by chen-keinan** [comment]
@erikgb you mention we lost some labels on Prev. change.  what are the missing labels? do we need them back?  can you please elaborate ?

**Comment by erikgb** [comment]
> @erikgb you mention we lost some labels on Prev. change. what are the missing labels? do we need them back? can you please elaborate ?

@chen-keinan After we started generating CRDs and clusterrole, we lost labels for the trivy-operator version and the source for the resources (trivy-operator). As a cluster-admin I prefer to know which version of the CRDs (especially) that are provisioned in my cluster. And because Helm does not natively support CRDs this is even more important - as Helm is currently our master format for resources.

**Comment by erikgb** [comment]
Closing as requested by @chen-keinan. But I would like to keep the linked issue open, as I would prefer to get labels back on CRDs and clusterrole.

**Comment by chen-keinan** [comment]
@erikgb I didn't requested to close this issue ,I ask if it can be closed as I wan't sure if it needed due to major refactoring on CRDs , it was open for long time and I wasn't aware that prev. changed removed important labels from CRDs

**Comment by erikgb** [comment]
> @erikgb I didn't requested to close this PR ,I asked `if it can be closed?` as I wan't sure if it needed due to major refactoring on CRDs , it was open for long time and I wasn't aware that prev. changed removed important labels from CRDs

@chen-keinan I saw that you comment was a (leading) question and not a request. 😉 I am just trying to introduce common patterns/tools for developing controllers from upstream Kubernetes (SIGS). But "fighting in headwind" with you is not something I am interesting in doing. This PR has been ready for review almost from day one, so you can only ask yourself why it has been lying open for such a long time. And please don't blame me for not reading PR descriptions:

> Sadly, the labels currently added to the CRDs (version and managed-by) cannot be sustained with this change, as this is an eschewed feature in controller-gen: https://github.com/kubernetes-sigs/controller-tools/issues/454. We should be able to get them back if https://github.com/aquasecurity/trivy-operator/discussions/239 lands on a migration to kustomize as the master format.

From https://github.com/aquasecurity/trivy-operator/pull/279

**Comment by erikgb** [comment]
> Anyway , this PR can be re-open to address the issues caused by CRDs refactoring for reviewal and approval.

Feel free to pick up anything I have done, including unmerged stuff. I claim no copyright on the code/changes, and all branches are still in my fork.

I am not motivated to involve in this project as it is managed at present. I'll find it more interesting to work on community driven projects and not a "one man show" like it appears to me now.

**Pull request closed without merging**